python/update_package_properties.py

```diff
@@ -0,0 +1,301 @@
+    if not re.match(r"v\d+\.\d+\.\d+$", tag):
+        raise ValueError(
+            "tag  should start with v and should include 3 levels of versions consists of numbers " +
+            "separated with dots. e.g: v10.0.1")
+
+
+@parameter_validation
+def is_email(email: str):
+    if not string_utils.is_email(email):
+        raise ValueError("Parameter is not in email format")
+
+
+@parameter_validation
+def is_project_changelog_header(header: str):
+    if header is None or not header:
+        raise ValueError("header should be non-empty and should not be None")
+    if not re.match(r"^### \w+\sv\d+\.\d+\.\d+\s\(\w+\s\d+,\s\d+\)\s###$", header):
+        raise ValueError(
+            f"changelog header is in invalid format. Actual:{header} Expected: ### citus v8.3.3 (March 23, 2021) ### ")
+
+
+def get_spec_file_name(project_name: str) -> str:
+    return f"{project_name}.spec"
+
+
+def get_version_number(version: str, fancy: bool, fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}{fancy_suffix}"
+
+
+def get_version_number_with_project_name(project_name: str, version: str, fancy: bool, fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}.{project_name}{fancy_suffix}"
+
+
+def get_template_environment(template_dir: str) -> Environment:
+    file_loader = FileSystemLoader(template_dir)
+    env = Environment(loader=file_loader)
+    return env
+
+
+def find_nth_overlapping(subject_string, search_string, n) -> int:
+    start = subject_string.find(search_string)
+
+    while start >= 0 and n > 1:
+        start = subject_string.find(search_string, start + 1)
+        n -= 1
+    return start
+
+
+def find_nth_overlapping_line_by_regex(subject_string, regex_pattern, n) -> int:
+    lines = subject_string.splitlines()
+    counter = 0
+    index = -1
+    for i in range(len(lines)):
+        if re.match(regex_pattern, lines[i]):
+            counter = counter + 1
+        if counter == n:
+            index = i
+            break
+    return index
+
+
+def get_last_changelog_content(all_changelog_content: str) -> str:
+    second_changelog_index = find_nth_overlapping(all_changelog_content, "###", 3)
+    changelogs = all_changelog_content[:second_changelog_index]
+    lines = changelogs.splitlines()
+    if len(lines) < 1:
+        raise ValueError("At least one line should be in changelog")
+    changelog_header = lines[0]
+    if not changelog_header.startswith("###"):
+        raise ValueError("Changelog header should start with '###'")
+    return changelogs
+
+
+def get_last_changelog_content_from_debian(all_changelog_content: str) -> str:
+    second_changelog_index = find_nth_overlapping_line_by_regex(all_changelog_content, "^[a-zA-Z]", 2)
+    lines = all_changelog_content.splitlines()
+    changelogs = "".join([str(elem) + "\n" for elem in lines[:second_changelog_index - 1]])
```

Comment:
```suggestion
    changelogs = "\n".join(lines[:second_changelog_index - 1])
```

Reply:
Fixed as you suggested with a small change
```python
changelogs = "\n".join(lines[:second_changelog_index - 1]) + "\n"
```